### PR TITLE
docs: Add generated API references to our book

### LINKED ISSUE #269: Update how we display our CRDs
**What would you like to be documented**:
We currently use https://doc.crds.dev/github.com/oracle/cluster-api-provider-oci to have a nice UI for reading our CRDs. This tool may be going away https://github.com/crdsdev/doc/issues/172.

We should probably pull the docs into our book like the Azure provider did https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3534

**Why is this needed**:
If the crds.dev tool is shutdown we won't have a nice way to users to read our CRDs.

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
If crds.dev goes away we have the APIs being generated in our book as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #269

### CODE REVIEW

**Comment by joekr** [comment]
I'm still not 100% sure if we want to commit the `raw` generated files or ignore them. I'm leaning towards committing them but I want to test how the book generation works in github actions.

Quick example of the generated docs 
image

File: docs/src/reference/v1beta2-exp-api.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# v1beta2 exp API
+
+{{ #include v1beta2-exp-api-raw.html }}
```
Comment (shyamradhakrishnan) [inline docs/src/reference/v1beta2-exp-api.md:3]: when we commit this, this particular file wont exists right? Is there a need to commit this file? Or is it that when the book job runs and the files will be generated and included in the book? If yes, then we are ok

Comment (joekr) [reply docs/src/reference/v1beta2-exp-api.md:3]: yeah from my above comment

> I'm still not 100% sure if we want to commit the raw generated files or ignore them. I'm leaning towards committing them but I want to test how the book generation works in github actions.

**Comment by shyamradhakrishnan** [comment]
Looks good, can you please paste output of unit test and build.sh before merging.

**Review by shyamradhakrishnan — APPROVED** [review]

**Pull request merged**